core/common/src/main/java/alluxio/util/CommonUtils.java

```diff
@@ -538,6 +538,8 @@ public static boolean isLocalHost(WorkerNetAddress address) {
 
   /**
    * Closes the netty channel from outside the netty I/O thread.
+   * NOTE: Be careful when holding any lock that can be acquired in the netty I/O thread when
+   * calling this function to avoid having deadlocks.
```

Comment:
I've checked all other calls. They are safe.  I don't know how to make this avoidable. Maybe more annotations and deadlock analysis?